crossopt/assignment3/unittests.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import unittest
+from assignment3.dictionary import Dictionary
+from assignment3.wordsearch import Grid
+from assignment3.wordsearch import findMatch, findMatches
+
+
+class TestDictionary(unittest.TestCase):
+    def test_isprefix(self):
+        dictionary = Dictionary(['CAR', 'CART', 'CARD', 'CAT'])
+        self.assertEqual(dictionary.isPrefix('A'), 0)
+        self.assertEqual(dictionary.isPrefix('car'), 0)
+        self.assertEqual(dictionary.isPrefix('CARDS'), 0)
+        self.assertEqual(dictionary.isPrefix(''), 1)
+        self.assertEqual(dictionary.isPrefix('CARD'), 1)
+        self.assertEqual(dictionary.isPrefix('CAR'), 1)
+
+    def test_isword(self):
+        dictionary = Dictionary(['CAR', 'CART', 'CARD', 'CAT'])
+        self.assertEqual(dictionary.isWord('A'), 0)
+        self.assertEqual(dictionary.isWord('C'), 0)
+        self.assertEqual(dictionary.isWord('car'), 0)
+        self.assertEqual(dictionary.isWord('CARDS'), 0)
+        self.assertEqual(dictionary.isWord(''), 0)
+        self.assertEqual(dictionary.isWord('CARD'), 1)
+        self.assertEqual(dictionary.isWord('CAR'), 1)
+
+
+class TestGrid(unittest.TestCase):
+    def test_getitem(self):
+        array = [['A', 'A', 'R'], ['T', 'C', 'D']]
+        grid = Grid(array)
+        self.assertEqual(grid.n, 2)
+        self.assertEqual(grid.m, 3)
+        for i in range(grid.n):
+            for j in range(grid.m):
+                self.assertEqual(grid[i][j], array[i][j])
+
+    def test_neighbors(self):
+        grid = Grid([['A', 'A', 'R'], ['T', 'C', 'D']])
+        self.assertSetEqual(set(grid.neighbors(0, 0)),
```

Comment:
Why do you transform the list of neighbors into a set?
It looks to me that the neighbors() function returns a stable order, and the set is not needed. Is this correct?

Reply:
Yes, in hindsight that was a bad idea.
When I was writing the tests I assumed that since the only thing I needed from the neighbors function was to return all the cell's neighbors in any order, I didn't need to check in which order they were actually returned